### PR TITLE
[IMP] add '--distinct-oca' flag to have a disinction for third party addons coming from OCA

### LINKED ISSUE #2: OCA category
By doing a HEAD request on `https://wheelhouse.odoo-community.org/oca-simple/odoo-addon-{addon-name}/`, we can determine if an addon is provided by OCA, and extract the OCA addons out of the "Third-party" category into their own category.

This could be enabled with an option, because this requires network access.

This would benefit from a local cache.

### PR DESCRIPTION
**Pull request by AnizR**
fixes #2 

> determine if an addon is provided by OCA, and extract the OCA addons out of the "Third-party" category into their own category

Note that:
- addons in merge request won't have OCA category
- it needs a connection to work

### CODE REVIEW

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py [hunk 1]
```diff
@@ -7,11 +7,17 @@
...
 from manifestoo_core.addon import Addon, is_addon_dir
 from manifestoo_core.core_addons import is_core_ce_addon, is_core_ee_addon
 from manifestoo_core.odoo_series import OdooSeries
+from requests import head
 
 NAME_DEFAULT_CATEGORY = "Default"
+OCA_ADDON_URL = "https://wheelhouse.odoo-community.org/oca-simple/odoo-addon-{addon_name}/"
+REQUEST_TIMEOUT = 2  # s
+
+other_addons_category_cache = Cache()
```
Comment (AnizR) [inline src/odoo_sort_manifest_depends/sort_manifest_deps.py]: Precise Cache position to reuse Cache between executions

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py [hunk 2]
```diff
@@ -7,11 +7,17 @@
...
 import click
 from click import command, option
+from diskcache import Cache
 from manifestoo_core.addon import Addon, is_addon_dir
 from manifestoo_core.core_addons import is_core_ce_addon, is_core_ee_addon
 from manifestoo_core.odoo_series import OdooSeries
+from requests import head
 
 NAME_DEFAULT_CATEGORY = "Default"
+OCA_ADDON_URL = "https://wheelhouse.odoo-community.org/oca-simple/odoo-addon-{addon_name}/"
```
Comment (sbidoul) [inline src/odoo_sort_manifest_depends/sort_manifest_deps.py]:
```suggestion
OCA_INDEX_URL= "https://wheelhouse.odoo-community.org/oca-simple/"
```

The use manifestoo-core `addon_name_to_distribution_name`

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py (hunk 1 above)
Comment (sbidoul) [inline src/odoo_sort_manifest_depends/sort_manifest_deps.py]: The project name (odoo-sort-manifest-depends) should somehow be in the disk cache location

File: README.md [hunk 3]
```diff
@@ -21,6 +21,8 @@ Options:
   --odoo-version TEXT           Project's Odoo version (e.g. 16.0)  [required]
   --project-name TEXT           Name of the project, will be the name of
                                 category of local addons (default: Local)
+  --distinct-oca                Add disctinction for third party addons coming
+                                from OCA
```
Comment (sbidoul) [inline README.md]: We'll need a `--reset-cache` to force a cache purge.

Comment (sbidoul) [reply README.md]: Or --refresh-cache to update cached entries.

File: README.md [hunk 4]
```diff
@@ -21,6 +21,8 @@ Options:
   --odoo-version TEXT           Project's Odoo version (e.g. 16.0)  [required]
   --project-name TEXT           Name of the project, will be the name of
                                 category of local addons (default: Local)
+  --distinct-oca                Add disctinction for third party addons coming
```
Comment (sbidoul) [inline README.md]: Not sure about the option name.

Comment (AnizR) [reply README.md]: I wasn't inspired,  `--oca-cateogry` seems better

File: README.md (hunk 3 above)
Comment (AnizR) [reply README.md]: I prefer the option `--reset-cache` to clear cache before execution

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py (hunk 2 above)
Comment (AnizR) [reply src/odoo_sort_manifest_depends/sort_manifest_deps.py]: I think that I spotted a small bug in the function `addon_name_to_distribution_name` : https://github.com/acsone/manifestoo-core/pull/78

File: README.md (hunk 4 above)
Comment (AnizR) [inline README.md]: To modify

**Comment by AnizR** [comment]
@FrancoMaxime ping

File: README.md [hunk 5]
```diff
@@ -21,6 +21,8 @@ Options:
   --odoo-version TEXT           Project's Odoo version (e.g. 16.0)  [required]
   --project-name TEXT           Name of the project, will be the name of
                                 category of local addons (default: Local)
+  --oca-cateogry                Add category for third party addons coming from OCA
```
Comment (FrancoMaxime) [inline README.md]: typo
```suggestion
  --oca-category                Add category for third party addons coming from OCA
```

Comment (AnizR) [reply README.md]: Well spotted, thank you!

**Comment by AnizR** [comment]
@lmignon ping

**Review by lmignon — APPROVED** [review]
GREAT!!!

**Comment by sbidoul** [comment]
Now we just need a little test. In another PR, it's fine too.

**Review by FrancoMaxime — APPROVED** [review]

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py [hunk 6]
```diff
@@ -7,11 +7,18 @@
...
 from click import command, option
+from diskcache import Cache
 from manifestoo_core.addon import Addon, is_addon_dir
 from manifestoo_core.core_addons import is_core_ce_addon, is_core_ee_addon
+from manifestoo_core.metadata import addon_name_to_distribution_name
 from manifestoo_core.odoo_series import OdooSeries
+from requests import head
 
 NAME_DEFAULT_CATEGORY = "Default"
+OCA_ADDON_URL = "https://wheelhouse.odoo-community.org/oca-simple/"
```
Comment (sbidoul) [inline src/odoo_sort_manifest_depends/sort_manifest_deps.py]:
```suggestion
OCA_ADDONS_INDEX_URL = "https://wheelhouse.odoo-community.org/oca-simple/"
```

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py [hunk 7]
```diff
@@ -7,11 +7,18 @@
...
+REQUEST_TIMEOUT = 2  # s
+
+other_addons_category_cache = Cache("~/.cache/odoo-sort-manifest-depends")
```
Comment (sbidoul) [inline src/odoo_sort_manifest_depends/sort_manifest_deps.py]: Use [platformdirs.user_cache_dir](https://pypi.org/project/platformdirs/) for portability.

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py [hunk 8]
```diff
@@ -34,7 +41,30 @@ def _get_addons_by_name(addons_dir: Path) -> dict[str, Addon]:
...
+
+    with other_addons_category_cache as cache:
+        for addon_name in addon_names:
+            if category := cache.get(addon_name):
+                if category == "oca":
+                    oca_addons.append(addon_name)
+                else:
+                    other_addons.append(addon_name)
+            else:
+                distribution_name = addon_name_to_distribution_name(addon_name.replace("_", "-"), odoo_series)
```
Comment (sbidoul) [inline src/odoo_sort_manifest_depends/sort_manifest_deps.py]:
```suggestion
                distribution_name = addon_name_to_distribution_name(addon_name, odoo_series).replace("_", "-")
```

File: src/odoo_sort_manifest_depends/sort_manifest_deps.py [hunk 9]
```diff
@@ -34,7 +41,30 @@ def _get_addons_by_name(addons_dir: Path) -> dict[str, Addon]:
...
+def _identify_oca_addons(addon_names: list[str], odoo_series: OdooSeries) -> tuple[list[str], list[str]]:
+    oca_addons, other_addons = [], []
+
+    with other_addons_category_cache as cache:
+        for addon_name in addon_names:
+            if category := cache.get(addon_name):
+                if category == "oca":
+                    oca_addons.append(addon_name)
+                else:
+                    other_addons.append(addon_name)
```
Comment (sbidoul) [inline src/odoo_sort_manifest_depends/sort_manifest_deps.py]: DRY this part by putting it after the category computation.

**Comment by AnizR** [comment]
> Now we just need a little test. In another PR, it's fine too.

It is on my todo list, I'll do it in a different PR

**Pull request merged**